File: fixtures/scale/python/large/cipherlab/ciphers/stridecode.py
"""Self-inverse codec (stridecode): reverse chunks of 2."""


def stridecode_encode(text):
    k = 2
    chunks = [text[i:i + k] for i in range(0, len(text), k)]
    return "".join(chunk[::-1] for chunk in chunks)


def stridecode_decode(text):
    return stridecode_encode(text)

File: fixtures/scale/typescript/large/src/ciphers/Stridecode.ts
// Self-inverse codec (stridecode): reverse chunks of 2.
export class Stridecode {
  static encode(text: string): string {
    const k = 2;
    const parts: string[] = [];
    for (let i = 0; i < text.length; i += k) {
      parts.push(text.slice(i, i + k).split('').reverse().join(''));
    }
    return parts.join('');
  }

  static decode(text: string): string {
    return Stridecode.encode(text);
  }
}
